cmd/topicctl/subcmd/apply.go

```diff
@@ -326,26 +283,23 @@ func applyTopic(
 			// if one of the steps after updateSettings errors when updating a topic,
 			// we can be in a state where some (but not all) changes were applied
 			// some topic creation errors also still create the topic
-			changesToBePrinted = unpackChanges(topicChanges, changesToBePrinted)
-			if changesToBePrinted.NewTopics != nil || changesToBePrinted.UpdatedTopics != nil {
+			if topicChanges != nil {
 				log.Error("Error detected while creating or updating a topic, the following changes were still made:")
-				partialChanges, printErr := printChanges(changesToBePrinted)
+				partialChanges, printErr := topicChanges.PrintJson()
 				if printErr != nil {
 					log.Error("Error printing partial JSON changes data")
 				} else {
 					log.Errorf("%#v", partialChanges)
 				}
-				return err
 			}
-		}
-		changesToBePrinted = unpackChanges(topicChanges, changesToBePrinted)
-	}
-
-	// ensure we're not printing empty json if there's no changes to the topic
-	if applyConfig.jsonOutput && (changesToBePrinted.NewTopics != nil || changesToBePrinted.UpdatedTopics != nil) {
-		if _, err := printChanges(changesToBePrinted); err != nil {
 			return err
 		}
+		// ensure we're not printing empty json if there's no changes to the topic
+		if applyConfig.jsonOutput && topicChanges != nil {
```

Comment:
Have you tested that sending an empty input to the notification script works fine? 
The notification script cannot fail if there is no change.

Reply:
yes, empty input to the notification script does nothing